### PR TITLE
Add 'recordings list' subcommand

### PR DESCRIPTION
**Pull request by wkalt**
Add capability to list and filter on recordings.

### CODE REVIEW

File: foxglove/console/client.go [hunk 1]
```diff
@@ -280,10 +280,14 @@ func (c *FoxgloveClient) DeleteExtension(id string) error {
 }
 
 func (c *FoxgloveClient) get(endpoint string, req any, target any) error {
-	querystring, err := form.EncodeToString(req)
+	buf := &bytes.Buffer{}
+	encoder := form.NewEncoder(buf)
+	encoder.DelimitWith('/')
```
Comment (james-rms) [inline foxglove/console/client.go:285]: worth a comment for why we do this

File: foxglove/console/api.go [hunk 2]
```diff
@@ -145,6 +145,90 @@ func (r AttachmentsResponse) Headers() []string {
...
+		"Metadata",
+	}
+}
+
+func (r RecordingsResponse) Fields() []string {
+	metadata, _ := json.Marshal(r.Metadata)
+	return []string{
+		r.ID,
+		r.Path,
+		fmt.Sprintf("%d", r.Size),
```
Comment (james-rms) [inline foxglove/console/api.go:215]:
```suggestion
		fmt.Sprint(r.Size),
```

File: foxglove/console/api.go [hunk 3]
```diff
@@ -145,6 +145,90 @@ func (r AttachmentsResponse) Headers() []string {
...
+	EdgeSite     SiteSummary      `json:"edgeSite"`
+	Device       DeviceSummary    `json:"device"`
+	Metadata     []MetadataRecord `json:"metadata"`
+}
+
+func (r RecordingsResponse) Headers() []string {
+	return []string{
+		"Recording ID",
+		"Path",
+		"Size",
```
Comment (james-rms) [inline foxglove/console/api.go:193]: units? we could do a human-readable bytes thing here?

Comment (wkalt) [reply foxglove/console/api.go:193]: sure, I can do that

File: foxglove/console/api.go (hunk 2 above)
Comment (james-rms) [reply foxglove/console/api.go:215]: Possibly [humanize](https://github.com/dustin/go-humanize) this?

File: foxglove/console/api.go [hunk 4]
```diff
@@ -145,6 +145,90 @@ func (r AttachmentsResponse) Headers() []string {
 	}
 }
 
+type RecordingsRequest struct {
+	DeviceID     string `json:"device.id" form:"device.id,omitempty"`
+	Start        string `json:"start" form:"start,omitempty"`
+	End          string `json:"end" form:"end,omitempty"`
+	Path         string `json:"path" form:"path,omitempty"`
+	SiteID       string `json:"siteId" form:"siteId,omitempty"`
```
Comment (james-rms) [inline foxglove/console/api.go:153]: why are these camelcase?
```suggestion
	SiteID       string `json:"site.id" form:"site.id,omitempty"`
```

File: foxglove/cmd/recordings.go [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+			)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "Failed to list recordings: %s\n", err)
+				os.Exit(1)
+			}
+		},
+	}
+	recordingsListCmd.InheritedFlags()
+	recordingsListCmd.PersistentFlags().StringVarP(&deviceID, "device-id", "", "", "Device ID")
+	recordingsListCmd.PersistentFlags().StringVarP(&start, "start", "", "", "start of data range")
```
Comment (james-rms) [inline foxglove/cmd/recordings.go:71]: as a user i don't really see how the command wants this formatted - maybe an example would help
```suggestion
	recordingsListCmd.PersistentFlags().StringVarP(&start, "start", "", "", "start of data range (eg. 2021-11-02T00:00:00Z)"
```

**Review by james-rms — APPROVED** [review]
This (among other resources) produces extremely wide terminal output when listing in the default table view. We have an improvement ticket (todo link) to allow the user to specify which columns they want, which would solve this problem generally.  With that in mind, i'm OK with this merging.

**Review by james-rms — APPROVED** [review]
This (among other resources) produces extremely wide terminal output when listing in the default table view. We have an improvement ticket (todo link) to allow the user to specify which columns they want, which would solve this problem generally.  With that in mind, i'm OK with this merging.

**Comment by wkalt** [comment]
here's the ticket: https://linear.app/foxglove/issue/FG-3529/foxglove-cli-handle-wide-outputs

**Pull request merged**